mvm/machine/signer.go

```diff
@@ -13,39 +15,79 @@ import (
 	"github.com/drand/kyber/sign/tbls"
 )
 
+const (
+	SignTypeTBLS      = 1
+	SignTypeSECP256K1 = 2
+)
+
+func (m *Machine) getProcess(processId string) *Process {
+	m.procLock.RLock()
+	defer m.procLock.RUnlock()
+	process, ok := m.processes[processId]
+	if !ok {
+		return nil
+	}
+	return process
+}
+
+func (m *Machine) GetLastSendTime(id string) time.Time {
+	m.smLock.Lock()
+	defer m.smLock.Unlock()
+	return m.sm[id]
```

Comment:
I don't understand why we need this `sm` in Machine, I think it's better to just make it a local state like before

Reply:
I have observed some repeated signatures in the MTG message group.

`sm` is a local variable used by the following two methods before:
`loopSignGroupEvents`
`loopReceiveGroupMessages`
I think we should unify the two `sm` in one to avoid sending repeated signatures.  What do you think?

Reply:
I don't think this is related to repeated messages. Because in these two functions, they send out different kind of messages, so they can't repeat the other one.

Reply:
There is a case of a same message with the same signature, but different threshold if we don't unify the two sm.

Reply:
It's possible that a MVM node gather enough signatures in `loopReceiveGroupMessages` before receiving the same MTG event in `loopSignGroupEvents `, in this case, two messages will be sent.